### ISSUE TITLE
Method of linear probing

### ISSUE THREAD

**Issue by Jason-Khan** [facebookresearch/mae#70]
Hi,

During linear probe training, all model parameters are frozen except the cls head. 
```
# freeze all but the head
    for _, p in model.named_parameters():
        p.requires_grad = False
    for _, p in model.head.named_parameters():
        p.requires_grad = True
```

So, the corresponding output of CLS token is considered as the representation of the image, and is used for linear probing classification.
But during pretraining, CLS token is never trained with respect to any objectives. How do we know the corresponding output of CLS is extracting a useful representation?
Shouldn't we also allow gradient to flow through the CLS token, so it can learn to pay attention to outputs of the ViT?

Also, is there a performance of global pooling for feature extraction instead of CLS token?

**Comment by LinB203** [comment]
anyone tried it?

**Comment by tim-heuwinkel** [comment]
I'm a bit late to the party but in the arXiv version of the paper the comment on this in appendix A.1: "As ViT has a class token [16], to adapt to this design, in our MAE pre-training we append an auxiliary dummy token to the encoder input. This token will be treated as the class token for training the classifier in linear probing and fine-tuning. Our MAE works similarly well without this token (with average pooling)."
